src/cpp/rtps/builtin/data/ProxyHashTables.hpp

```diff
@@ -51,10 +52,18 @@ class node_segregator
             std::size_t nodes_to_allocate,
             const bool& flag,
             std::size_t padding = foonathan::memory::detail::memory_block_stack::implementation_offset)
-        : block_size_(nodes_to_allocate * node_size + padding)
-        , node_allocator_(new allocator_type(node_size, nodes_to_allocate * node_size + padding))
+        : block_size_(nodes_to_allocate * (node_size
```

Comment:
Should we take into account the `memory_pool::min_node_size` ?

Reply:
Done